internal/grpc/services/gateway/usershareprovider.go

```diff
@@ -336,7 +340,7 @@ func (s *svc) UpdateReceivedShare(ctx context.Context, req *collaboration.Update
 	}, nil
 }
 
-func (s *svc) createReference(ctx context.Context, resourceID *provider.ResourceId) *rpc.Status {
+func (s *svc) createReference(ctx context.Context, resourceID *provider.ResourceId, delete string) *rpc.Status {
```

Comment:
What are the semantics of the `delete` parameter?

Reply:
ohh I see, it signals the storageprovider to remove the reference that we just created? I think I need more context to fully grasp this one ^^

Reply:
That was just a hacky way to implement the Deletion without having a `RemoveReference` method in the CS3 API. This is not the finished code. :sweat_smile:

Reply:
oooooooooohh oh I see, sorry I was too fast on having a look at this while in wip